### PR TITLE
Hts 868

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala [hunk 1]
```diff
@@ -68,5 +78,99 @@ object CreateAccountRequestValidator {
...
+   */
+  private def specialCharacters(s: String, ignore: List[Char] = List.empty[Char]): List[Char] =
+    s.replaceAllLiterally(" ", "").filter(isSpecial(_, ignore)).toList.distinct
+
+  /** Does the given string contain `n` or more consecutive special characters? */
+  private def containsNConsecutiveSpecialCharacters(s: String, n: Int): Boolean =
+    containsNConsecutive(s, n, isSpecial(_))
+
+  /** Does the given string contain an apostrophe? */
+  private def containAnApostrophe(s: String): Boolean =
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala]: it debatable whether or not you need this as a separate def as it's so short and it's only being called in one place

File: test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala [hunk 2]
```diff
@@ -104,6 +105,36 @@ class CreateAccountRequestValidatorSpec extends TestSupport {
...
+        }
+
+        "has a forename containing an apostrophe" in {
+          val result = validator.validateRequest(
+            CreateAccountRequest(
+              validCreateAccountHeader,
+              validCreateAccountBody.copy(forename = "fore'name")
+            ))
+
+          result.isInvalid shouldBe true
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala]: you could do something like
```scala
result shouldBe Invalid(NonEmptylist.of("blah blah"))
```
or you could even do:
```scala
result.isInvalid shouldBe true

Comment (AndyHWChung) [reply test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala]: you could even have:
```scala
def testIsInvalid(createAccountRequest: CreateAccountRequest): Unit = validator.validateRequest(createAccountRequest).isInvalid shouldBe true
```

File: app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala]: ok I've removed the separate method

File: test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala (hunk 2 above)
Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala]: I believe I tried to do this before and I got this error:

Invalid(NonEmptyList(forename contains an apostrophe)) was not equal to Invalid(NonEmptyList(forename contains an apostrophe, ))

It seems impossible to create a NonEmptyList with only one parameter so it expects two

File: app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidator.scala]: done, thanks

File: test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala (hunk 2 above)
Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosaveapi/validators/CreateAccountRequestValidatorSpec.scala]: done, thanks

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**